<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>2 Download RNA-Seq data from GEO | From FASTQ files to Variant Calling for RNA-Seq</title>
  <meta name="description" content="2 Download RNA-Seq data from GEO | From FASTQ files to Variant Calling for RNA-Seq">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="2 Download RNA-Seq data from GEO | From FASTQ files to Variant Calling for RNA-Seq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Download RNA-Seq data from GEO | From FASTQ files to Variant Calling for RNA-Seq" />
  
  
  

<meta name="author" content="Anna Quaglieri">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="3-ch-downsample.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>1.1</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-download.html"><a href="2-ch-download.html"><i class="fa fa-check"></i><b>2</b> Download RNA-Seq data from GEO</a><ul>
<li class="chapter" data-level="2.1" data-path="2-ch-download.html"><a href="2-ch-download.html#get-srx-sample-names"><i class="fa fa-check"></i><b>2.1</b> Get SRX sample names</a></li>
<li class="chapter" data-level="2.2" data-path="2-ch-download.html"><a href="2-ch-download.html#create-ncbi-query"><i class="fa fa-check"></i><b>2.2</b> Create NCBI query</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-downsample.html"><a href="3-ch-downsample.html"><i class="fa fa-check"></i><b>3</b> Downsampling FASTQ files</a></li>
<li class="chapter" data-level="4" data-path="4-ch-input-pipe.html"><a href="4-ch-input-pipe.html"><i class="fa fa-check"></i><b>4</b> Define files and programs needed for the pipeline</a></li>
<li class="chapter" data-level="5" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html"><i class="fa fa-check"></i><b>5</b> FASTQC and adapters trimming</a><ul>
<li class="chapter" data-level="5.1" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#parallelise-your-fastqc"><i class="fa fa-check"></i><b>5.1</b> Parallelise your <code>FASTQC</code></a></li>
<li class="chapter" data-level="5.2" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#summarise-reports-with-multiqc"><i class="fa fa-check"></i><b>5.2</b> Summarise reports with <code>MultiQC</code></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-align.html"><a href="6-ch-align.html"><i class="fa fa-check"></i><b>6</b> Alignement, Read Groups, Mark duplicates</a><ul>
<li class="chapter" data-level="6.1" data-path="6-ch-align.html"><a href="6-ch-align.html#create-star-index"><i class="fa fa-check"></i><b>6.1</b> Create <code>STAR</code> index</a></li>
<li class="chapter" data-level="6.2" data-path="6-ch-align.html"><a href="6-ch-align.html#star-1pass"><i class="fa fa-check"></i><b>6.2</b> STAR-1pass</a></li>
<li class="chapter" data-level="6.3" data-path="6-ch-align.html"><a href="6-ch-align.html#star-2pass"><i class="fa fa-check"></i><b>6.3</b> STAR-2pass</a></li>
<li class="chapter" data-level="6.4" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:post-align"><i class="fa fa-check"></i><b>6.4</b> Details about post-alignment functions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="7-ch-merge-bamfiles.html"><a href="7-ch-merge-bamfiles.html"><i class="fa fa-check"></i><b>7</b> Merge bamfiles</a></li>
<li class="chapter" data-level="8" data-path="8-ch-gatk-prep.html"><a href="8-ch-gatk-prep.html"><i class="fa fa-check"></i><b>8</b> GATK pre-processing</a><ul>
<li class="chapter" data-level="8.1" data-path="8-ch-gatk-prep.html"><a href="8-ch-gatk-prep.html#splitncigarreads"><i class="fa fa-check"></i><b>8.1</b> SplitNCigarReads</a></li>
<li class="chapter" data-level="8.2" data-path="8-ch-gatk-prep.html"><a href="8-ch-gatk-prep.html#base-recalibration"><i class="fa fa-check"></i><b>8.2</b> Base recalibration</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html"><i class="fa fa-check"></i><b>9</b> Variant calling</a><ul>
<li class="chapter" data-level="9.1" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html#call-with-mutect2"><i class="fa fa-check"></i><b>9.1</b> Call with MuTect2</a></li>
<li class="chapter" data-level="9.2" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html#call-with-samtools-varscan2"><i class="fa fa-check"></i><b>9.2</b> Call with Samtools + VarScan2</a></li>
<li class="chapter" data-level="9.3" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html#call-with-vardict"><i class="fa fa-check"></i><b>9.3</b> Call with VarDict</a></li>
<li class="chapter" data-level="9.4" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html#annotate-variants"><i class="fa fa-check"></i><b>9.4</b> Annotate variants</a></li>
<li class="chapter" data-level="9.5" data-path="9-ch-call-variants.html"><a href="9-ch-call-variants.html#indels-km"><i class="fa fa-check"></i><b>9.5</b> Targeted INDEL calling with <code>km</code> algorithm</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-ch-variants-st.html"><a href="10-ch-variants-st.html"><i class="fa fa-check"></i><b>10</b> Standardise output of variants</a><ul>
<li class="chapter" data-level="10.1" data-path="10-ch-variants-st.html"><a href="10-ch-variants-st.html#superfreq-to-combine-time-course-mutations-for-one-patient"><i class="fa fa-check"></i><b>10.1</b> superFreq to combine time-course mutations for one patient</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-ch-fusions.html"><a href="11-ch-fusions.html"><i class="fa fa-check"></i><b>11</b> Call fusions with <code>STAR-Fusion</code> and <code>FusionInspector</code></a></li>
<li class="chapter" data-level="12" data-path="12-ch-superfreq.html"><a href="12-ch-superfreq.html"><i class="fa fa-check"></i><b>12</b> Analyse time-course mutations with superFreq</a><ul>
<li class="chapter" data-level="12.1" data-path="12-ch-superfreq.html"><a href="12-ch-superfreq.html#sec:mutexplore"><i class="fa fa-check"></i><b>12.1</b> Explore mutations with the <code>Mutexplore</code> Shiny app</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From FASTQ files to Variant Calling for RNA-Seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:download" class="section level1">
<h1><span class="header-section-number">2</span> Download RNA-Seq data from GEO</h1>
<p>We will provide an example on how to download data from GEO using the Leucegene CBF-AML RNS-Seq data uploaded at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49642" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49642</a>. The first step in dowloading data from GEO is to download the <code>SRA</code> files. For that we need to get <code>SRX</code> entries which each one corresponds to a sample in the RNA-Seq cohort.</p>
<div id="get-srx-sample-names" class="section level2">
<h2><span class="header-section-number">2.1</span> Get SRX sample names</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GEOquery)
<span class="kw">library</span>(tidyverse)
<span class="kw">library</span>(knitr)
<span class="kw">library</span>(stringr)</code></pre></div>
<p>Below is an example using one accession number from the <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49642">Leucegene data</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Get matrix files for every accession number</span>
series_matrix_info &lt;-<span class="st"> </span><span class="cf">function</span>(gse){
gsed &lt;-<span class="st"> </span><span class="kw">getGEO</span>(gse,<span class="dt">GSEMatrix=</span><span class="ot">TRUE</span>)
gse.mat &lt;-<span class="st"> </span><span class="kw">pData</span>(<span class="kw">phenoData</span>(gsed[[<span class="dv">1</span>]]))
reduced &lt;-<span class="st"> </span>gse.mat[,<span class="kw">c</span>(<span class="st">&quot;title&quot;</span>,<span class="st">&quot;geo_accession&quot;</span>,<span class="st">&quot;relation.1&quot;</span>)]
<span class="kw">write.csv</span>(reduced,<span class="kw">file.path</span>(<span class="st">&quot;../data&quot;</span>,<span class="kw">paste</span>(gse,<span class="st">&quot;_&quot;</span>,<span class="kw">nrow</span>(gse.mat),<span class="st">&quot;.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)),<span class="dt">row.names =</span> <span class="ot">FALSE</span>)
}

<span class="kw">series_matrix_info</span>(<span class="st">&quot;GSE49642&quot;</span>) <span class="co"># 43 samples</span></code></pre></div>
<p>Every row in Table <a href="2-ch-download.html#tab:SRX">2.1</a> contains sample names (<code>title</code>) and <code>GSM</code> accession numbers. In order to download a particular sample we need the <code>SRA</code> terms which are the names starting with: <code>SRX***</code> in the <code>relation.1</code> column. The structure of the matrix might change across different studies but you should be able to find <code>SRX</code> entries hidden somewhere in the <code>GSEMatrix</code>!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">matrix_file &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="dt">path =</span> <span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>),<span class="dt">pattern =</span> <span class="st">&quot;GSE&quot;</span>,<span class="dt">full.names =</span> <span class="ot">TRUE</span>)
GSEmatrix &lt;-<span class="st"> </span><span class="kw">read_csv</span>(matrix_file)

<span class="kw">kable</span>(GSEmatrix[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,],<span class="dt">caption=</span><span class="st">&quot;SRX sample names.&quot;</span>)</code></pre></div>
<table>
<caption><span id="tab:SRX">Table 2.1: </span>SRX sample names.</caption>
<thead>
<tr class="header">
<th align="left">title</th>
<th align="left">geo_accession</th>
<th align="left">relation.1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">02H053</td>
<td align="left">GSM1203305</td>
<td align="left">SRA: <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRX332625" class="uri">https://www.ncbi.nlm.nih.gov/sra?term=SRX332625</a></td>
</tr>
<tr class="even">
<td align="left">02H066</td>
<td align="left">GSM1203306</td>
<td align="left">SRA: <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRX332626" class="uri">https://www.ncbi.nlm.nih.gov/sra?term=SRX332626</a></td>
</tr>
<tr class="odd">
<td align="left">03H041</td>
<td align="left">GSM1203307</td>
<td align="left">SRA: <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRX332627" class="uri">https://www.ncbi.nlm.nih.gov/sra?term=SRX332627</a></td>
</tr>
<tr class="even">
<td align="left">03H116</td>
<td align="left">GSM1203308</td>
<td align="left">SRA: <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRX332628" class="uri">https://www.ncbi.nlm.nih.gov/sra?term=SRX332628</a></td>
</tr>
<tr class="odd">
<td align="left">03H119</td>
<td align="left">GSM1203309</td>
<td align="left">SRA: <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRX332629" class="uri">https://www.ncbi.nlm.nih.gov/sra?term=SRX332629</a></td>
</tr>
</tbody>
</table>
<p>With some string processing we can extract the <code>SRX</code> entries.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GSEmatrix<span class="op">$</span>SRX &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_extract</span>(<span class="dt">string =</span> GSEmatrix<span class="op">$</span>relation.<span class="dv">1</span>,<span class="dt">pattern =</span> <span class="st">&quot;SRX[0-9][0-9][0-9][0-9][0-9][0-9]&quot;</span>)
GSEmatrix<span class="op">$</span>relation.<span class="dv">1</span> &lt;-<span class="st"> </span><span class="ot">NULL</span>
<span class="kw">kable</span>(<span class="kw">head</span>(GSEmatrix))</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">title</th>
<th align="left">geo_accession</th>
<th align="left">SRX</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">02H053</td>
<td align="left">GSM1203305</td>
<td align="left">SRX332625</td>
</tr>
<tr class="even">
<td align="left">02H066</td>
<td align="left">GSM1203306</td>
<td align="left">SRX332626</td>
</tr>
<tr class="odd">
<td align="left">03H041</td>
<td align="left">GSM1203307</td>
<td align="left">SRX332627</td>
</tr>
<tr class="even">
<td align="left">03H116</td>
<td align="left">GSM1203308</td>
<td align="left">SRX332628</td>
</tr>
<tr class="odd">
<td align="left">03H119</td>
<td align="left">GSM1203309</td>
<td align="left">SRX332629</td>
</tr>
<tr class="even">
<td align="left">04H024</td>
<td align="left">GSM1203310</td>
<td align="left">SRX332630</td>
</tr>
</tbody>
</table>
</div>
<div id="create-ncbi-query" class="section level2">
<h2><span class="header-section-number">2.2</span> Create NCBI query</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">search_ncbi &lt;-<span class="st"> </span><span class="kw">paste</span>(GSEmatrix<span class="op">$</span>SRX,<span class="dt">collapse=</span><span class="st">&quot; OR &quot;</span>)
search_ncbi</code></pre></div>
<pre><code>## [1] &quot;SRX332625 OR SRX332626 OR SRX332627 OR SRX332628 OR SRX332629 OR SRX332630 OR SRX332631 OR SRX332632 OR SRX332633 OR SRX332634 OR SRX332635 OR SRX332636 OR SRX332637 OR SRX332638 OR SRX332639 OR SRX332640 OR SRX332641 OR SRX332642 OR SRX332643 OR SRX332644 OR SRX332645 OR SRX332646 OR SRX332647 OR SRX332648 OR SRX332649 OR SRX332650 OR SRX332651 OR SRX332652 OR SRX332653 OR SRX332654 OR SRX332655 OR SRX332656 OR SRX332657 OR SRX332658 OR SRX332659 OR SRX332660 OR SRX332661 OR SRX332662 OR SRX332663 OR SRX332664 OR SRX332665 OR SRX332666 OR SRX332667&quot;</code></pre>
<p>Paste the search SRX332625 OR SRX332626 OR SRX332627 OR SRX332628 OR SRX332629 OR SRX332630 OR SRX332631 OR SRX332632 OR SRX332633 OR SRX332634 OR SRX332635 OR SRX332636 OR SRX332637 OR SRX332638 OR SRX332639 OR SRX332640 OR SRX332641 OR SRX332642 OR SRX332643 OR SRX332644 OR SRX332645 OR SRX332646 OR SRX332647 OR SRX332648 OR SRX332649 OR SRX332650 OR SRX332651 OR SRX332652 OR SRX332653 OR SRX332654 OR SRX332655 OR SRX332656 OR SRX332657 OR SRX332658 OR SRX332659 OR SRX332660 OR SRX332661 OR SRX332662 OR SRX332663 OR SRX332664 OR SRX332665 OR SRX332666 OR SRX332667 into NCBI <a href="https://www.ncbi.nlm.nih.gov/sra" class="uri">https://www.ncbi.nlm.nih.gov/sra</a> and follow the intructions in <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sradownload/#download-sequence-data-files-usi" class="uri">https://www.ncbi.nlm.nih.gov/sra/docs/sradownload/#download-sequence-data-files-usi</a> <strong>Download sequence data files using SRA Toolkit</strong> to download all the <code>SRR</code> run names and information of the runs.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># Files are saved in the home directory under ncbi/public/sra</span>
<span class="ex">prefetch</span> --option-file SraAccList_CBF-AML_Leucegene.txt</code></pre></div>
<p>SRA files can then be converted to <code>fastq</code> files with <code>fastq-dump --split-files</code>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="3-ch-downsample.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["RNA-variant-book.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
