<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>8 Variant calling | From FASTQ files to Variant Calling for RNA-Seq</title>
  <meta name="description" content="8 Variant calling | From FASTQ files to Variant Calling for RNA-Seq">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="8 Variant calling | From FASTQ files to Variant Calling for RNA-Seq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Variant calling | From FASTQ files to Variant Calling for RNA-Seq" />
  
  
  

<meta name="author" content="Anna Quaglieri">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="7-ch-gatk-prep.html">
<link rel="next" href="9-ch-variants-st.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./">RNA-Seq variant calling</a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>1.2</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-download.html"><a href="2-ch-download.html"><i class="fa fa-check"></i><b>2</b> Download RNA-Seq data from GEO</a><ul>
<li class="chapter" data-level="2.1" data-path="2-ch-download.html"><a href="2-ch-download.html#get-srx-sample-names"><i class="fa fa-check"></i><b>2.1</b> Get SRX sample names</a></li>
<li class="chapter" data-level="2.2" data-path="2-ch-download.html"><a href="2-ch-download.html#create-ncbi-query"><i class="fa fa-check"></i><b>2.2</b> Create NCBI query</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-downsample.html"><a href="3-ch-downsample.html"><i class="fa fa-check"></i><b>3</b> Downsampling FASTQ or BAM files</a></li>
<li class="chapter" data-level="4" data-path="4-ch-input-pipe.html"><a href="4-ch-input-pipe.html"><i class="fa fa-check"></i><b>4</b> Define files and programs needed for the pipeline</a></li>
<li class="chapter" data-level="5" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html"><i class="fa fa-check"></i><b>5</b> FASTQC and adapters trimming</a><ul>
<li class="chapter" data-level="5.1" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#parallelise-your-fastqc"><i class="fa fa-check"></i><b>5.1</b> Parallelise your <code>FASTQC</code></a></li>
<li class="chapter" data-level="5.2" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#summarise-reports-with-multiqc"><i class="fa fa-check"></i><b>5.2</b> Summarise reports with <code>MultiQC</code></a></li>
<li class="chapter" data-level="5.3" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#adapter-troubles-star-vs-subread"><i class="fa fa-check"></i><b>5.3</b> Adapter troubles: STAR vs Subread</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-align.html"><a href="6-ch-align.html"><i class="fa fa-check"></i><b>6</b> Alignement, Read Groups, Mark duplicates</a><ul>
<li class="chapter" data-level="6.1" data-path="6-ch-align.html"><a href="6-ch-align.html#create-star-index"><i class="fa fa-check"></i><b>6.1</b> Create <code>STAR</code> index</a></li>
<li class="chapter" data-level="6.2" data-path="6-ch-align.html"><a href="6-ch-align.html#star-1pass"><i class="fa fa-check"></i><b>6.2</b> STAR-1pass</a></li>
<li class="chapter" data-level="6.3" data-path="6-ch-align.html"><a href="6-ch-align.html#star-2pass"><i class="fa fa-check"></i><b>6.3</b> STAR-2pass</a></li>
<li class="chapter" data-level="6.4" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:post-align"><i class="fa fa-check"></i><b>6.4</b> Details about post-alignment functions</a></li>
<li class="chapter" data-level="6.5" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:merge-bamfiles"><i class="fa fa-check"></i><b>6.5</b> Merge bamfiles</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html"><i class="fa fa-check"></i><b>7</b> GATK pre-processing</a><ul>
<li class="chapter" data-level="7.1" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#splitncigarreads"><i class="fa fa-check"></i><b>7.1</b> SplitNCigarReads</a></li>
<li class="chapter" data-level="7.2" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#base-recalibration"><i class="fa fa-check"></i><b>7.2</b> Base recalibration</a></li>
<li class="chapter" data-level="7.3" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#summary-with-multiqc"><i class="fa fa-check"></i><b>7.3</b> Summary with MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html"><i class="fa fa-check"></i><b>8</b> Variant calling</a><ul>
<li class="chapter" data-level="8.1" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-mutect2"><i class="fa fa-check"></i><b>8.1</b> Call with MuTect2</a></li>
<li class="chapter" data-level="8.2" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-samtools-varscan2"><i class="fa fa-check"></i><b>8.2</b> Call with Samtools + VarScan2</a></li>
<li class="chapter" data-level="8.3" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-vardict"><i class="fa fa-check"></i><b>8.3</b> Call with VarDict</a></li>
<li class="chapter" data-level="8.4" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#annotate-variants"><i class="fa fa-check"></i><b>8.4</b> Annotate variants</a></li>
<li class="chapter" data-level="8.5" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#indels-km"><i class="fa fa-check"></i><b>8.5</b> Targeted INDEL calling with <code>km</code> algorithm</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html"><i class="fa fa-check"></i><b>9</b> Standardise output of variants</a><ul>
<li class="chapter" data-level="9.1" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html#superfreq-to-combine-time-course-mutations-for-one-patient"><i class="fa fa-check"></i><b>9.1</b> superFreq to combine time-course mutations for one patient</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-ch-fusions.html"><a href="10-ch-fusions.html"><i class="fa fa-check"></i><b>10</b> Call fusions with <code>STAR-Fusion</code> and <code>FusionInspector</code></a></li>
<li class="chapter" data-level="11" data-path="11-ch-superfreq.html"><a href="11-ch-superfreq.html"><i class="fa fa-check"></i><b>11</b> Analyse time-course mutations with superFreq</a><ul>
<li class="chapter" data-level="11.1" data-path="11-ch-superfreq.html"><a href="11-ch-superfreq.html#sec:mutexplore"><i class="fa fa-check"></i><b>11.1</b> Explore mutations with the <code>Mutexplore</code> Shiny app</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-ch-refer.html"><a href="12-ch-refer.html"><i class="fa fa-check"></i><b>12</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From FASTQ files to Variant Calling for RNA-Seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:call-variants" class="section level1">
<h1><span class="header-section-number">8</span> Variant calling</h1>
<p>You can perform variant calling on the whole genome or on specific regions that you can specify with a <code>.bed</code> file.
Here we show how to call variants on specific regions of interest. The target regions were created using the gene symbols of the mutations listed in <strong>Supplemental Table 3</strong> of <span class="citation">(Lavallée et al. <a href="12-ch-refer.html#ref-Lavallee2016-sf">2016</a>)</span> and listed in <code>./data/mutations_Lavallee_2016.csv</code>. We used the hg19 inbuilt annotation of <code>Rsubread</code> to obtain the gene ranges and added 500 bp at the end and at the beginning of each gene. The final <code>bed</code> files is <code>./data/target_regions.bed</code>.</p>
<p>View all the options in the <code>./functions/call_variants.R</code> function.</p>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">Rscript</span> ./functions/call_variants.R --help</code></pre>
<p>Below is an example on how to call variants with <code>VarDict</code> and <code>VarScan</code> in tumour-only mode and annotate them with the <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Preditor (VEP)</a>. The directory needed for VarDict can be downloaded from the software GitHub page <a href="https://github.com/AstraZeneca-NGS/VarDict" class="uri">https://github.com/AstraZeneca-NGS/VarDict</a>.</p>
<p>For other caller options:</p>
<pre><code>
-c CHARACTER, --variant_caller=CHARACTER
                Variant caller to be used. Possible values are &#39;mutect&#39;, &#39;varscan&#39;, &#39;freebayes&#39; and &#39;vardict&#39;. Samtools mpileup is used to create VarScan input.
                </code></pre>
<p>The function above is a wrapper for the calls done by each caller. Below are details of the actual caller settings that I normally use to call variants with these callers.</p>
<div id="call-with-mutect2" class="section level2">
<h2><span class="header-section-number">8.1</span> Call with MuTect2</h2>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">module</span> load gatk/3.7.0

<span class="ex">gatk</span> -T MuTect2 -R path_to_genome.fa \
-I:tumor ./results/star_2pass/SRR1608907_Aligned.reorderedDupl.rg.split.recalibrated.bam \
-L ./data/target_regions.bed \
-o ./results/mutect/regions/SampleName_germline_snvs_indels.vcf \
-log ./results/mutect/regions/SampleName_germline_snvs_indels_log</code></pre>
</div>
<div id="call-with-samtools-varscan2" class="section level2">
<h2><span class="header-section-number">8.2</span> Call with Samtools + VarScan2</h2>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">module</span> load varscan/2.3.9
<span class="ex">module</span> load samtools/1.6

<span class="ex">samtools</span> mpileup --output-tags AD,ADF,ADR,DP,SP \
--fasta-ref path_to_genome.fa \
-l ./data/target_regions.bed ./results/star_2pass/SRR1608907_Aligned.reorderedDupl.rg.split.recalibrated.bam <span class="kw">|</span> <span class="ex">varscan</span> mpileup2cns --variants 1 --output-vcf 1 --min-var-freq 0.01 <span class="op">&gt;</span> ./results/varscan/regions/SampleName_germline_snvs_indels.vcf <span class="op">2&gt;</span> ./results/varscan/regions/SampleName_germline_snvs_indels_log</code></pre>
</div>
<div id="call-with-vardict" class="section level2">
<h2><span class="header-section-number">8.3</span> Call with VarDict</h2>
<p>The <code>teststrandbias.R</code> and <code>var2vcf_valid.pl</code> scripts were downloaded from <a href="https://github.com/AstraZeneca-NGS/VarDict" class="uri">https://github.com/AstraZeneca-NGS/VarDict</a>. VarDict only calls variants on a target region.</p>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">module</span> load vardict/1.5.1

<span class="ex">vardict</span> -f 0.05 -c 1 -S 2 -E 3 -g 4 -r 2 -t -th 10 -v -G  path_to_genome.fa \
-b ./results/star_2pass/SRR1608907_Aligned.reorderedDupl.rg.bam ./data/target_regions.bed  <span class="kw">|</span> <span class="ex">vardict_dir/teststrandbias.R</span> <span class="kw">|</span> <span class="ex">vardict_dir/var2vcf_valid.pl</span> -N -E -f 0.05 <span class="op">&gt;</span>  ./results/vardict/regions/SampleName_germline_snvs_indels.vcf</code></pre>
</div>
<div id="annotate-variants" class="section level2">
<h2><span class="header-section-number">8.4</span> Annotate variants</h2>
<p>This step is performed within the <code>call_variants.R</code> function when the path to the VEP cache library is provided.</p>
<pre><code>--VEPcall &#39;vep --dir_cache path_to_vep_cache/.vep --offline&#39; \</code></pre>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">module</span> load ensembl-vep/89.0

<span class="ex">vep</span> --dir_cache dir_to_VEP_cache/.vep --offline \
-i ./results/varscan/regions/SampleName_germline_snvs_indels.vcf \
-o ./results/varscan/regions/annotated_variants/SampleName_germline_annotated.vcf \
--cache --everything --force_overwrite --assembly GRCh37 --fork 12 --vcf --port 3337</code></pre>
</div>
<div id="indels-km" class="section level2">
<h2><span class="header-section-number">8.5</span> Targeted INDEL calling with <code>km</code> algorithm</h2>
<p>The <a href="https://github.com/iric-soft/km"><code>km</code> algorithm</a> <span class="citation">(Eric Olivier Audemard, Patrick Gendron, Vincent-Philippe Lavallée, Josée Hébert, Guy Sauvageau, Sébastien Lemieux <a href="12-ch-refer.html#ref-Software-km">2018</a>)</span> is one of the sofware developed for targeted INDEL calls in RNA-Seq. Calling INDELs from RNA-Seq is challenging and several tools have been published between 2017-2019 to accomplish this task. The <code>km</code> algorithm was benchmarked on the AML Leucegene and TCGA-LAML datasets.</p>
<p>As shown in Figure <a href="index.html#fig:overview">1.1</a> I used it together with VarDict to create a set of INDELs calls from leukemia RNA-Seq. In my approach, I call variants with both callers and curate their ouput by checking INDELs that are called by both callers (only one calls is preserved) or by only one call. In my experience, their calls are pretty similar and I was able to call FLT3 Internal Tandem Duplications on my samples using both callers.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="7-ch-gatk-prep.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="9-ch-variants-st.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/annaquaglieri16/RNA-seq-variant-calling/tree/master/Rmd/09-Call-annotate-variants.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["RNA-Variants-book.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
