<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>9 varikondo and superFreq: Standardise and combine variants over time | From FASTQ files to Variant Calling for RNA-Seq</title>
  <meta name="description" content="9 varikondo and superFreq: Standardise and combine variants over time | From FASTQ files to Variant Calling for RNA-Seq">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="9 varikondo and superFreq: Standardise and combine variants over time | From FASTQ files to Variant Calling for RNA-Seq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 varikondo and superFreq: Standardise and combine variants over time | From FASTQ files to Variant Calling for RNA-Seq" />
  
  
  

<meta name="author" content="Anna Quaglieri">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="8-ch-call-variants.html">
<link rel="next" href="10-ch-fusions.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./">RNA-Seq variant calling</a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>1.2</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-download.html"><a href="2-ch-download.html"><i class="fa fa-check"></i><b>2</b> Download RNA-Seq data from GEO</a><ul>
<li class="chapter" data-level="2.1" data-path="2-ch-download.html"><a href="2-ch-download.html#get-srx-sample-names"><i class="fa fa-check"></i><b>2.1</b> Get SRX sample names</a></li>
<li class="chapter" data-level="2.2" data-path="2-ch-download.html"><a href="2-ch-download.html#create-ncbi-query"><i class="fa fa-check"></i><b>2.2</b> Create NCBI query</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-downsample.html"><a href="3-ch-downsample.html"><i class="fa fa-check"></i><b>3</b> Downsampling FASTQ or BAM files</a></li>
<li class="chapter" data-level="4" data-path="4-ch-input-pipe.html"><a href="4-ch-input-pipe.html"><i class="fa fa-check"></i><b>4</b> Define files and programs needed for the pipeline</a></li>
<li class="chapter" data-level="5" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html"><i class="fa fa-check"></i><b>5</b> FASTQC and adapters trimming</a><ul>
<li class="chapter" data-level="5.1" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#parallelise-your-fastqc"><i class="fa fa-check"></i><b>5.1</b> Parallelise your <code>FASTQC</code></a></li>
<li class="chapter" data-level="5.2" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#summarise-reports-with-multiqc"><i class="fa fa-check"></i><b>5.2</b> Summarise reports with <code>MultiQC</code></a></li>
<li class="chapter" data-level="5.3" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#adapter-troubles-star-vs-subread"><i class="fa fa-check"></i><b>5.3</b> Adapter troubles: STAR vs Subread</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-align.html"><a href="6-ch-align.html"><i class="fa fa-check"></i><b>6</b> Alignement, Read Groups, Mark duplicates</a><ul>
<li class="chapter" data-level="6.1" data-path="6-ch-align.html"><a href="6-ch-align.html#create-star-index"><i class="fa fa-check"></i><b>6.1</b> Create <code>STAR</code> index</a></li>
<li class="chapter" data-level="6.2" data-path="6-ch-align.html"><a href="6-ch-align.html#star-1pass"><i class="fa fa-check"></i><b>6.2</b> STAR-1pass</a></li>
<li class="chapter" data-level="6.3" data-path="6-ch-align.html"><a href="6-ch-align.html#star-2pass"><i class="fa fa-check"></i><b>6.3</b> STAR-2pass</a></li>
<li class="chapter" data-level="6.4" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:post-align"><i class="fa fa-check"></i><b>6.4</b> Details about post-alignment functions</a></li>
<li class="chapter" data-level="6.5" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:merge-bamfiles"><i class="fa fa-check"></i><b>6.5</b> Merge bamfiles</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html"><i class="fa fa-check"></i><b>7</b> GATK pre-processing</a><ul>
<li class="chapter" data-level="7.1" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#splitncigarreads"><i class="fa fa-check"></i><b>7.1</b> SplitNCigarReads</a></li>
<li class="chapter" data-level="7.2" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#base-recalibration"><i class="fa fa-check"></i><b>7.2</b> Base recalibration</a></li>
<li class="chapter" data-level="7.3" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#summary-with-multiqc"><i class="fa fa-check"></i><b>7.3</b> Summary with MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html"><i class="fa fa-check"></i><b>8</b> Variant calling</a><ul>
<li class="chapter" data-level="8.1" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-mutect2"><i class="fa fa-check"></i><b>8.1</b> Call with MuTect2</a></li>
<li class="chapter" data-level="8.2" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-samtools-varscan2"><i class="fa fa-check"></i><b>8.2</b> Call with Samtools + VarScan2</a></li>
<li class="chapter" data-level="8.3" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-vardict"><i class="fa fa-check"></i><b>8.3</b> Call with VarDict</a></li>
<li class="chapter" data-level="8.4" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#annotate-variants"><i class="fa fa-check"></i><b>8.4</b> Annotate variants</a></li>
<li class="chapter" data-level="8.5" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#indels-km"><i class="fa fa-check"></i><b>8.5</b> Targeted INDEL calling with <code>km</code> algorithm</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html"><i class="fa fa-check"></i><b>9</b> <code>varikondo</code> and <code>superFreq</code>: Standardise and combine variants over time</a><ul>
<li class="chapter" data-level="9.1" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html#parse_vcf_output-read-in-your-single-sample-vcf-file-in-r"><i class="fa fa-check"></i><b>9.1</b> <code>parse_vcf_output</code>: read in your single sample VCF file in R</a></li>
<li class="chapter" data-level="9.2" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html#superfreq-and-varikodnoimport_goi_superfreq-to-combine-snvsshort-insertions-over-time"><i class="fa fa-check"></i><b>9.2</b> <code>superFreq</code> and <code>varikodno::import_goi_superfreq</code> to combine SNVs/short insertions over time</a></li>
<li class="chapter" data-level="9.3" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html#import_any-combine-variants-over-time-for-any-caller"><i class="fa fa-check"></i><b>9.3</b> <code>import_any</code>: combine variants over time for any caller</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-ch-fusions.html"><a href="10-ch-fusions.html"><i class="fa fa-check"></i><b>10</b> Call fusions with <code>STAR-Fusion</code> and <code>FusionInspector</code></a></li>
<li class="chapter" data-level="11" data-path="11-ch-refer.html"><a href="11-ch-refer.html"><i class="fa fa-check"></i><b>11</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From FASTQ files to Variant Calling for RNA-Seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:variants-st" class="section level1">
<h1><span class="header-section-number">9</span> <code>varikondo</code> and <code>superFreq</code>: Standardise and combine variants over time</h1>
<p>We are now at the stage where variants are called and we have loads of <code>VCF</code> files/ exel files. The scripts in the previous sections helps with calling variants using 4 callers (VarDict, Varscan, MuTect2, km and freebayes).</p>
<p>At this stage, depending on how one needs to use the variants, there are several options discussed below. To make it easier to import variant calls into R and store them in data frame I developed the R package <a href="https://annaquaglieri16.github.io/varikondo/"><code>varikondo</code></a> which hopefully will help you to find the spark joy in your variants!</p>
<p>Below are some applications but for a more in depth description of its functionalities see the package website <a href="https://annaquaglieri16.github.io/varikondo/articles/vignette.html" class="uri">https://annaquaglieri16.github.io/varikondo/articles/vignette.html</a>.</p>
<div id="parse_vcf_output-read-in-your-single-sample-vcf-file-in-r" class="section level2">
<h2><span class="header-section-number">9.1</span> <code>parse_vcf_output</code>: read in your single sample VCF file in R</h2>
<p>If you simply want to read a VCF file in R you can use <code>parse_vcf_output</code> which currently support tumour-only variants from <code>GATK MuTect2</code>, <code>VarDict</code>, <code>VarScan</code> and <code>Freebayes</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(varikondo)

annot_vcf_mutect &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;germline_mutect.vcf&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;varikondo&quot;</span>)
annot_vcf_mutect</code></pre>
<pre><code>## [1] &quot;/Users/quaglieri.a/Library/R/3.5/library/varikondo/extdata/germline_mutect.vcf&quot;</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">parsed_vcf_mutect &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw">parse_vcf_output</span>(<span class="dt">vcf_path =</span> annot_vcf_mutect, 
                                                 <span class="dt">caller =</span> <span class="st">&quot;mutect&quot;</span>,
                                                 <span class="dt">sample_name =</span> <span class="st">&quot;Sample1&quot;</span>,
                                                 <span class="dt">vep =</span> <span class="ot">TRUE</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(parsed_vcf_mutect[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,],<span class="dt">caption =</span> <span class="st">&quot;Parsed MuTect2 output, from VCF to data frame.&quot;</span>)</code></pre>
<table>
<caption><span id="tab:parse-mutect">Table 9.1: </span>Parsed MuTect2 output, from VCF to data frame.</caption>
<thead>
<tr class="header">
<th align="left">Location</th>
<th align="left">caller</th>
<th align="left">chrom</th>
<th align="left">pos</th>
<th align="right">end</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="right">qual</th>
<th align="left">filter</th>
<th align="left">genotype</th>
<th align="right">tot_depth</th>
<th align="right">VAF</th>
<th align="right">ref_depth</th>
<th align="right">alt_depth</th>
<th align="right">ref_forw</th>
<th align="right">ref_rev</th>
<th align="right">alt_forw</th>
<th align="right">alt_rev</th>
<th align="right">start</th>
<th align="right">width</th>
<th align="right">ref_base_quality</th>
<th align="right">alt_base_quality</th>
<th align="left">SampleName</th>
<th align="left">Allele</th>
<th align="left">Consequence</th>
<th align="left">IMPACT</th>
<th align="left">SYMBOL</th>
<th align="left">Gene</th>
<th align="left">Feature_type</th>
<th align="left">Feature</th>
<th align="left">BIOTYPE</th>
<th align="left">EXON</th>
<th align="left">INTRON</th>
<th align="left">HGVSc</th>
<th align="left">HGVSp</th>
<th align="left">cDNA_position</th>
<th align="left">CDS_position</th>
<th align="left">Protein_position</th>
<th align="left">Amino_acids</th>
<th align="left">Codons</th>
<th align="left">Existing_variation</th>
<th align="left">DISTANCE</th>
<th align="left">STRAND</th>
<th align="left">FLAGS</th>
<th align="left">VARIANT_CLASS</th>
<th align="left">SYMBOL_SOURCE</th>
<th align="left">HGNC_ID</th>
<th align="left">CANONICAL</th>
<th align="left">TSL</th>
<th align="left">APPRIS</th>
<th align="left">CCDS</th>
<th align="left">ENSP</th>
<th align="left">SWISSPROT</th>
<th align="left">TREMBL</th>
<th align="left">UNIPARC</th>
<th align="left">GENE_PHENO</th>
<th align="left">SIFT</th>
<th align="left">PolyPhen</th>
<th align="left">DOMAINS</th>
<th align="left">AF</th>
<th align="left">AFR_AF</th>
<th align="left">AMR_AF</th>
<th align="left">EAS_AF</th>
<th align="left">EUR_AF</th>
<th align="left">SAS_AF</th>
<th align="left">AA_AF</th>
<th align="left">EA_AF</th>
<th align="left">ExAC_AF</th>
<th align="left">ExAC_Adj_AF</th>
<th align="left">ExAC_AFR_AF</th>
<th align="left">ExAC_AMR_AF</th>
<th align="left">ExAC_EAS_AF</th>
<th align="left">ExAC_FIN_AF</th>
<th align="left">ExAC_NFE_AF</th>
<th align="left">ExAC_OTH_AF</th>
<th align="left">ExAC_SAS_AF</th>
<th align="left">MAX_AF</th>
<th align="left">MAX_AF_POPS</th>
<th align="left">CLIN_SIG</th>
<th align="left">SOMATIC</th>
<th align="left">PHENO</th>
<th align="left">PUBMED</th>
<th align="left">MOTIF_NAME</th>
<th align="left">MOTIF_POS</th>
<th align="left">HIGH_INF_POS</th>
<th align="left">MOTIF_SCORE_CHANGE</th>
<th align="left">IMPACT_rank</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000331941</td>
<td align="left">protein_coding</td>
<td align="left">11/16</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1627</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS412.1</td>
<td align="left">ENSP00000332180</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000002AA5B</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.978)</td>
<td align="left">Superfamily_domains:SSF49265&amp;SMART_domains:SM00060&amp;Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;hmmpanther:PTHR23036&amp;hmmpanther:PTHR23036:SF75&amp;PROSITE_profiles:PS50853</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000338937</td>
<td align="left">protein_coding</td>
<td align="left">11/14</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1627</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000345013</td>
<td align="left"></td>
<td align="left">E9PK56&amp;E1B6W6</td>
<td align="left">UPI000013CA3F</td>
<td align="left">1</td>
<td align="left">deleterious(0.01)</td>
<td align="left">probably_damaging(0.987)</td>
<td align="left">Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;PROSITE_profiles:PS50853&amp;hmmpanther:PTHR23036&amp;hmmpanther:PTHR23036:SF75&amp;SMART_domains:SM00060&amp;Superfamily_domains:SSF49265</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000361632</td>
<td align="left">protein_coding</td>
<td align="left">12/16</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1689</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS413.1</td>
<td align="left">ENSP00000355406</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000004CAC4</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.992)</td>
<td align="left">PROSITE_profiles:PS50853&amp;hmmpanther:PTHR23036:SF75&amp;hmmpanther:PTHR23036&amp;Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;SMART_domains:SM00060&amp;Superfamily_domains:SSF49265</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000373103</td>
<td align="left">protein_coding</td>
<td align="left">13/17</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">2175</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left">YES</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS414.1</td>
<td align="left">ENSP00000362195</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000002AA5A</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.949)</td>
<td align="left">Superfamily_domains:SSF49265&amp;SMART_domains:SM00060&amp;Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;hmmpanther:PTHR23036:SF75&amp;hmmpanther:PTHR23036&amp;PROSITE_profiles:PS50853</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000373104</td>
<td align="left">protein_coding</td>
<td align="left">13/18</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">2175</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS412.1</td>
<td align="left">ENSP00000362196</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000002AA5B</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.978)</td>
<td align="left">Superfamily_domains:SSF49265&amp;SMART_domains:SM00060&amp;Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;hmmpanther:PTHR23036:SF75&amp;hmmpanther:PTHR23036&amp;PROSITE_profiles:PS50853</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000373106</td>
<td align="left">protein_coding</td>
<td align="left">13/17</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">2175</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS413.1</td>
<td align="left">ENSP00000362198</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000004CAC4</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.992)</td>
<td align="left">Superfamily_domains:SSF49265&amp;SMART_domains:SM00060&amp;Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;hmmpanther:PTHR23036&amp;hmmpanther:PTHR23036:SF75&amp;PROSITE_profiles:PS50853</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">upstream_gene_variant</td>
<td align="left">MODIFIER</td>
<td align="left">MRPS15</td>
<td align="left">ENSG00000116898</td>
<td align="left">Transcript</td>
<td align="left">ENST00000373116</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3734</td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">14504</td>
<td align="left">YES</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS411.1</td>
<td align="left">ENSP00000362208</td>
<td align="left">P82914</td>
<td align="left">D3DPS9</td>
<td align="left">UPI0000135287</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000418048</td>
<td align="left">protein_coding</td>
<td align="left">11/15</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1627</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS413.1</td>
<td align="left">ENSP00000401588</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000004CAC4</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.992)</td>
<td align="left">Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;PROSITE_profiles:PS50853&amp;hmmpanther:PTHR23036&amp;hmmpanther:PTHR23036:SF75&amp;SMART_domains:SM00060&amp;Superfamily_domains:SSF49265</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">missense_variant</td>
<td align="left">MODERATE</td>
<td align="left">CSF3R</td>
<td align="left">ENSG00000119535</td>
<td align="left">Transcript</td>
<td align="left">ENST00000440588</td>
<td align="left">protein_coding</td>
<td align="left">11/15</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1627</td>
<td align="left">1627</td>
<td align="left">543</td>
<td align="left">A/T</td>
<td align="left">Gca/Aca</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">2439</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS414.1</td>
<td align="left">ENSP00000397568</td>
<td align="left">Q99062</td>
<td align="left">E9PK56</td>
<td align="left">UPI000002AA5A</td>
<td align="left">1</td>
<td align="left">deleterious(0)</td>
<td align="left">probably_damaging(0.949)</td>
<td align="left">Gene3D:2.60.40.10&amp;Pfam_domain:PF00041&amp;PROSITE_profiles:PS50853&amp;hmmpanther:PTHR23036&amp;hmmpanther:PTHR23036:SF75&amp;SMART_domains:SM00060&amp;Superfamily_domains:SSF49265</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">chr1_36933772</td>
<td align="left">mutect2</td>
<td align="left">chr1</td>
<td align="left">36933772</td>
<td align="right">36933772</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">43.49469</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">532</td>
<td align="right">0.017</td>
<td align="right">523</td>
<td align="right">9</td>
<td align="right">261</td>
<td align="right">262</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">36933772</td>
<td align="right">1</td>
<td align="right">15571</td>
<td align="right">247</td>
<td align="left">Sample1</td>
<td align="left">T</td>
<td align="left">upstream_gene_variant</td>
<td align="left">MODIFIER</td>
<td align="left">MRPS15</td>
<td align="left">ENSG00000116898</td>
<td align="left">Transcript</td>
<td align="left">ENST00000462067</td>
<td align="left">processed_transcript</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3826</td>
<td align="left">-1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">14504</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
</tbody>
</table>
</div>
<div id="superfreq-and-varikodnoimport_goi_superfreq-to-combine-snvsshort-insertions-over-time" class="section level2">
<h2><span class="header-section-number">9.2</span> <code>superFreq</code> and <code>varikodno::import_goi_superfreq</code> to combine SNVs/short insertions over time</h2>
<p>If you have data collected for the same patient over time you can use software like <a href="https://github.com/ChristofferFlensburg/superFreq"><code>superFreq</code></a> to track changing clones over the course of treatment for example. Figure <a href="index.html#fig:overview">1.1</a> shows the input necessary for <code>superFreq</code> and full documentation/manuscript/examples are available on GitHub <a href="https://github.com/ChristofferFlensburg/superFreq" class="uri">https://github.com/ChristofferFlensburg/superFreq</a>. The chunk below is an example taken from one of the runs that I did for RNA samples, setting the <code>mode = RNA</code> option. There main input for <code>superFreq</code> are: VCF files (in my case normally from VarScan); bamfiles (including Panel Of Normal (PON) samples); and metadata information (see example in Table <a href="#tab:metadata"><strong>??</strong></a>).</p>
<p><code>superFreq</code> will create two main output folder specified when running the function: <code>plots = './superFreq/plots'</code> and <code>rdir = './superFreq/R'</code>.</p>
<p>You can then use <code>varikondo::import_goi_superfreq</code> to variants found on genes of interest into R!</p>
<pre class="sourceCode r"><code class="sourceCode r">import_sf &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw">import_goi_superfreq</span>(<span class="dt">superFreq_R_path =</span> superFreq_R_path,
                                 <span class="dt">superFreq_meta_path =</span> superFreq_meta_path,
                                 studyGenes,
                                 <span class="dt">patientID =</span> <span class="st">&quot;D1&quot;</span>,
                                 <span class="dt">ref_genome =</span> <span class="st">&quot;hg38&quot;</span>,
                                 <span class="dt">VAFcut =</span> <span class="fl">0.15</span>)</code></pre>
</div>
<div id="import_any-combine-variants-over-time-for-any-caller" class="section level2">
<h2><span class="header-section-number">9.3</span> <code>import_any</code>: combine variants over time for any caller</h2>
<p>In the pipeline (Figure <a href="index.html#fig:overview">1.1</a>) that I applied to the RNA-Seq cohorts that I was involved with, I cann INDELs with <code>VarDict</code> and <code>km</code> for every single sample independently and then filter and curate the results. <code>superFreq</code> only supports point mutations and short INDELs which is why I won’t use for more complex INDELs reported in VCF files by <code>VarDict</code>. My strategy to combine and filter variants within the same patient is to filter them simultaneusly and to retain only those variants which passes specific threshold at one time points. The rational being here that if a variant is too low to be independently detected at one time point it might still be kept if it is present at a good level at a different time.</p>
<p>To accomplish this, I use the <code>varikondo::import_any</code> function which takes in input a data frame, as created by <code>import_vcf_output</code> and filter variants by patient. Here is <a href="https://example-varikondo.netlify.com/">an example</a> on how to jointly use <code>import_vcf_output</code> and <code>import_any</code> to obtain a final filtered and combine version of VarDict calls for one patients with samples over time.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="8-ch-call-variants.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-ch-fusions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/annaquaglieri16/RNA-seq-variant-calling/tree/master/Rmd/10-Standardise-variants.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["RNA-Variants-book.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
