<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>9 Standardise output of variants | From FASTQ files to Variant Calling for RNA-Seq</title>
  <meta name="description" content="9 Standardise output of variants | From FASTQ files to Variant Calling for RNA-Seq">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="9 Standardise output of variants | From FASTQ files to Variant Calling for RNA-Seq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Standardise output of variants | From FASTQ files to Variant Calling for RNA-Seq" />
  
  
  

<meta name="author" content="Anna Quaglieri">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="8-ch-call-variants.html">
<link rel="next" href="10-ch-fusions.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><strong><a href="./">RNA-Seq variant calling</a></strong></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#disclaimer"><i class="fa fa-check"></i><b>1.2</b> Disclaimer</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-ch-download.html"><a href="2-ch-download.html"><i class="fa fa-check"></i><b>2</b> Download RNA-Seq data from GEO</a><ul>
<li class="chapter" data-level="2.1" data-path="2-ch-download.html"><a href="2-ch-download.html#get-srx-sample-names"><i class="fa fa-check"></i><b>2.1</b> Get SRX sample names</a></li>
<li class="chapter" data-level="2.2" data-path="2-ch-download.html"><a href="2-ch-download.html#create-ncbi-query"><i class="fa fa-check"></i><b>2.2</b> Create NCBI query</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-ch-downsample.html"><a href="3-ch-downsample.html"><i class="fa fa-check"></i><b>3</b> Downsampling FASTQ or BAM files</a></li>
<li class="chapter" data-level="4" data-path="4-ch-input-pipe.html"><a href="4-ch-input-pipe.html"><i class="fa fa-check"></i><b>4</b> Define files and programs needed for the pipeline</a></li>
<li class="chapter" data-level="5" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html"><i class="fa fa-check"></i><b>5</b> FASTQC and adapters trimming</a><ul>
<li class="chapter" data-level="5.1" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#parallelise-your-fastqc"><i class="fa fa-check"></i><b>5.1</b> Parallelise your <code>FASTQC</code></a></li>
<li class="chapter" data-level="5.2" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#summarise-reports-with-multiqc"><i class="fa fa-check"></i><b>5.2</b> Summarise reports with <code>MultiQC</code></a></li>
<li class="chapter" data-level="5.3" data-path="5-ch-fastqc.html"><a href="5-ch-fastqc.html#adapter-troubles-star-vs-subread"><i class="fa fa-check"></i><b>5.3</b> Adapter troubles: STAR vs Subread</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-ch-align.html"><a href="6-ch-align.html"><i class="fa fa-check"></i><b>6</b> Alignement, Read Groups, Mark duplicates</a><ul>
<li class="chapter" data-level="6.1" data-path="6-ch-align.html"><a href="6-ch-align.html#create-star-index"><i class="fa fa-check"></i><b>6.1</b> Create <code>STAR</code> index</a></li>
<li class="chapter" data-level="6.2" data-path="6-ch-align.html"><a href="6-ch-align.html#star-1pass"><i class="fa fa-check"></i><b>6.2</b> STAR-1pass</a></li>
<li class="chapter" data-level="6.3" data-path="6-ch-align.html"><a href="6-ch-align.html#star-2pass"><i class="fa fa-check"></i><b>6.3</b> STAR-2pass</a></li>
<li class="chapter" data-level="6.4" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:post-align"><i class="fa fa-check"></i><b>6.4</b> Details about post-alignment functions</a></li>
<li class="chapter" data-level="6.5" data-path="6-ch-align.html"><a href="6-ch-align.html#sec:merge-bamfiles"><i class="fa fa-check"></i><b>6.5</b> Merge bamfiles</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html"><i class="fa fa-check"></i><b>7</b> GATK pre-processing</a><ul>
<li class="chapter" data-level="7.1" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#splitncigarreads"><i class="fa fa-check"></i><b>7.1</b> SplitNCigarReads</a></li>
<li class="chapter" data-level="7.2" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#base-recalibration"><i class="fa fa-check"></i><b>7.2</b> Base recalibration</a></li>
<li class="chapter" data-level="7.3" data-path="7-ch-gatk-prep.html"><a href="7-ch-gatk-prep.html#summary-with-multiqc"><i class="fa fa-check"></i><b>7.3</b> Summary with MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html"><i class="fa fa-check"></i><b>8</b> Variant calling</a><ul>
<li class="chapter" data-level="8.1" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-mutect2"><i class="fa fa-check"></i><b>8.1</b> Call with MuTect2</a></li>
<li class="chapter" data-level="8.2" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-samtools-varscan2"><i class="fa fa-check"></i><b>8.2</b> Call with Samtools + VarScan2</a></li>
<li class="chapter" data-level="8.3" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#call-with-vardict"><i class="fa fa-check"></i><b>8.3</b> Call with VarDict</a></li>
<li class="chapter" data-level="8.4" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#annotate-variants"><i class="fa fa-check"></i><b>8.4</b> Annotate variants</a></li>
<li class="chapter" data-level="8.5" data-path="8-ch-call-variants.html"><a href="8-ch-call-variants.html#indels-km"><i class="fa fa-check"></i><b>8.5</b> Targeted INDEL calling with <code>km</code> algorithm</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html"><i class="fa fa-check"></i><b>9</b> Standardise output of variants</a><ul>
<li class="chapter" data-level="9.1" data-path="9-ch-variants-st.html"><a href="9-ch-variants-st.html#superfreq-to-combine-time-course-mutations-for-one-patient"><i class="fa fa-check"></i><b>9.1</b> superFreq to combine time-course mutations for one patient</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-ch-fusions.html"><a href="10-ch-fusions.html"><i class="fa fa-check"></i><b>10</b> Call fusions with <code>STAR-Fusion</code> and <code>FusionInspector</code></a></li>
<li class="chapter" data-level="11" data-path="11-ch-superfreq.html"><a href="11-ch-superfreq.html"><i class="fa fa-check"></i><b>11</b> Analyse time-course mutations with superFreq</a><ul>
<li class="chapter" data-level="11.1" data-path="11-ch-superfreq.html"><a href="11-ch-superfreq.html#sec:mutexplore"><i class="fa fa-check"></i><b>11.1</b> Explore mutations with the <code>Mutexplore</code> Shiny app</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-ch-refer.html"><a href="12-ch-refer.html"><i class="fa fa-check"></i><b>12</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Proudly published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From FASTQ files to Variant Calling for RNA-Seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ch:variants-st" class="section level1">
<h1><span class="header-section-number">9</span> Standardise output of variants</h1>
<p>We are now at the stage where variants are called. The scripts above helps with calling variants
using 4 callers (VarDict, Varscan, MuTect2, km).</p>
<p>At this stage, depending on how one needs to use the variants, there are several options:</p>
<div id="superfreq-to-combine-time-course-mutations-for-one-patient" class="section level2">
<h2><span class="header-section-number">9.1</span> superFreq to combine time-course mutations for one patient</h2>
<p>If you have time-course data for each patient you can use software like <a href="https://github.com/ChristofferFlensburg/superFreq"><code>superFreq</code></a> to analyse mutations over time; estimate the clonality development of your cancer samples; CNVs; somatic mutations etc…</p>
<p>If you decide to call variants with one of the callers above independently from the <code>call_variants.R</code> function, you still have options to standardise the <code>VCF</code> output from a caller so to explore it with the <code>Mutexplore</code> app.<br />
- No annotation is run -&gt; apply function from <code>samplepower</code>
- Annotation is run -&gt; apply function from <code>samplepower</code>
- Output is taken from superFreq -&gt; functions from <code>lineplots</code></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="8-ch-call-variants.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-ch-fusions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/annaquaglieri16/RNA-seq-variant-calling/tree/master/Rmd/10-Standardise-variants.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["RNA-Variants-book.pdf"],
"toc": {
"collapse": "none"
}
});
});
</script>

</body>

</html>
